### ISSUE TITLE
Cannot do forward pass: inputs not set

### ISSUE THREAD

**Issue by Joshuaalbert** [deeplearning4j/deeplearning4j#3333]
#### Issue Description
I'm doing regression with a computation graph and multidata set.
Upon geting to `cg.fit(dataset)` I get 
```
Cannot do forward pass: inputs not set
```
Likewise, I cannot run `cg.score(dataset)` as this gives the same error.
The network and example code is given here https://gist.github.com/Joshuaalbert/3df125a74c352fbc4c3ea15ea56f6360

#### Version Information
I using 8.1-SNAPSHOT

**Comment by AlexDBlack** [comment]
This is most likely due to a misconfigured network, that isn't picked up correctly during validation for some reason. For example, a layer that doesn't have an input, doesn't feed into any other layer, or something similar.

I had a look at your code, but couldn't see an obvious cause - though it's a bit hard to follow the structure there.

**Comment by tomthetrainer** [comment]
I wonder if outputing the model config to JSON would provide insight into the issue?

**Comment by Joshuaalbert** [comment]
@tomthetrainer what function does this? I'm only aware of the model serializer.

**Comment by tomthetrainer** [comment]
ComputationGraphConfiguration.toJSON

So in your code, looks like cgconf.toJSON

**Comment by Joshuaalbert** [comment]
Great! Easy to find the problem. I found one merge vertex has two inputs from the same layer.
Thanks
